File: Question 0001 - Two Sum/src/Solution.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];

        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int x = target - nums[i];

            if (map.containsKey(x)) {
                ans[0] = i;
                ans[1] = map.get(x);
                return ans;
            }

            map.put(ans[i], i);
        }

        return ans;
    }
}

File: Question 0001 - Two Sum/src/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

typedef unordered_map<int, int> umii;

#define pb push_back

class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<int> ans;
        umii map;
        int n = nums.size();

        for(int i = 0; i < n; ++i) map[nums[i]] = i;

        for(int i = 0; i < n; i++)
        {
            int comp = target - nums[i];
            if(map.find(comp) != map.end() && map[comp] != i)
            {
                ans.pb(i);
                ans.pb(map[comp]);
                break;
            }

        }

        return ans;
    }
};